Cilsil.Test/E2E/InferSharpTest.cs

```diff
@@ -776,6 +776,29 @@ public void NullExceptionIsInst(int testInputCode, InferError expectedError)
                     DerefObject(VarName.FirstLocal), GetString(expectedError));
         }
 
+        /// <summary>
+        /// Validates thread safety violation detection of a public method's read without
+        /// synchronization on an integer field which may race with a write on that integer field.
+        /// </summary>
+        /// <param name="encloseReadInLock"><c>true</c> if the read should be enclosed in a lock,
+        /// <c>false</c> otherwise.</param>
+        /// <param name="expectedError">The expected error.</param>
+        [DataRow(false, InferError.THREAD_SAFETY_VIOLATION)]
+        [DataRow(true, InferError.None)]
+        [DataTestMethod]
+        public void ThreadSafetyViolationSimple(bool encloseReadInLock, InferError expectedError)
```

Comment:
Yes. I have pulled exception handling modules from main

Reply:
And this PR is targeting main branch of my fork, not the main branch in main repo

Reply:
Hmmm, in that case can we wait until those modifications are in before committing this? If we commit this and then base off of master I think we see all of the modifications together right?

Reply:
> Hmmm, in that case can we wait until those modifications are in before committing this? If we commit this and then base off of master I think we see all of the modifications together right?

Sure. I am just trying to separate changes. So it would be easier to review.